pkg/util/image_util.go

```diff
@@ -107,11 +107,25 @@ func remoteImage(image string, opts *config.KanikoOptions) (v1.Image, error) {
 	}
 
 	registryName := ref.Context().RegistryStr()
+	var newReg name.Registry
+
+	if opts.RegistryMirror != "" && registryName == name.DefaultRegistry {
```

Comment:
this seems to break the following situation:
DefaultRegistry: index.docker.io
mirror-registry unset
Dockerfile starts with from localhost:5000/xxx

localhost:5000 does not require authentication, however following error is seen: Error while retrieving image from cache: localhost:5000/kaniko-test/onbuild-base:latest UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:kaniko-test/onbuild-base Type:repository]] 

See here for more info: https://travis-ci.org/GoogleContainerTools/kaniko/builds/644019948#L3547

Reply:
@samos123 I've managed to reproduce this locally and fix it. Thanks.